flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/data/FlinkParquetReaders.java

```diff
@@ -411,6 +416,43 @@ public DecimalData read(DecimalData ignored) {
     }
   }
 
+  private static class NanosToTimestampReader
+      extends ParquetValueReaders.UnboxedReader<TimestampData> {
+    NanosToTimestampReader(ColumnDescriptor desc) {
+      super(desc);
+    }
+
+    @Override
+    public TimestampData read(TimestampData ignored) {
+      long value = readLong();
+      return TimestampData.fromInstant(Instant.ofEpochSecond(0, value));
```

Comment:
How could we cross check, that the Spark write/Flink read and Flink write/Spark read is correctly working?
Are we sure that this is the expected data format?

Reply:
Maybe the generic java writer tests?